modules/openapi-generator/src/main/resources/swift5/model_doc.mustache

```diff
@@ -3,7 +3,7 @@
 ## Properties
 Name | Type | Description | Notes
 ------------ | ------------- | ------------- | -------------
-{{#vars}}**{{{name}}}** | {{#isPrimitiveType}}**{{{dataType}}}**{{/isPrimitiveType}}{{^isPrimitiveType}}{{^isContainer}}[**{{dataType}}**]({{complexType}}.md){{/isContainer}}{{#isContainer}}{{{dataType}}}{{/isContainer}}{{/isPrimitiveType}} | {{description}} | {{^required}}[optional] {{/required}}{{#isReadOnly}}[readonly] {{/isReadOnly}}{{#defaultValue}}[default to {{{.}}}]{{/defaultValue}}
+{{#vars}}**{{{name}}}** | {{#isPrimitiveType}}**{{{dataType}}}**{{/isPrimitiveType}}{{^isPrimitiveType}}{{^isContainer}}[**{{dataType}}**]({{complexType}}.md){{/isContainer}}{{#isContainer}}{{{dataType}}}{{/isContainer}}{{/isPrimitiveType}} | {{description}} | {{^required}}[optional] {{/required}}{{#isReadOnly}}[readonly] {{/isReadOnly}}{{#defaultValue}}[default to {{#vendorExtensions.x-null-encodable}}.encodeValue({{{.}}}){{/vendorExtensions.x-null-encodable}}{{^vendorExtensions.x-null-encodable}}{{{.}}}{{/vendorExtensions.x-null-encodable}}]{{/defaultValue}}
```

Comment:
@jarrodparkes here should we check if the default value is null and put .encondeNull?

Reply:
@4brunu I'm not sure what is different about this template. From this template, it is like I am unable to access {{{x-null-encodable-default-value}}}. Maybe this template is resolved prior to the `postProcessModels` function call?

I'm not sure there is a mustache check I can make which checks IF defaultValue == null.

Reply:
Hum, I see, since this is in the docs, I don't think it's too bad.

Reply:
@4brunu Is there some sort of `preProcessModels` method? I'm just wondering if I am on to something about when this template is resolved in relation to the `postProcessModels` function

Reply:
weird... looking into `DefaultGenerator` it seems like `postProcessModels` would have already been called...

https://github.com/OpenAPITools/openapi-generator/blob/611a3effa60d32b59c9402766a87d144ecf1eec4/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L507#L541

```java
        allProcessedModels = config.postProcessAllModels(allProcessedModels);


        // generate files based on processed models
        for (String modelName : allProcessedModels.keySet()) {
            Map<String, Object> models = (Map<String, Object>) allProcessedModels.get(modelName);
            models.put("modelPackage", config.modelPackage());
            try {
                //don't generate models that have an import mapping
                if (config.importMapping().containsKey(modelName)) {
                    continue;
                }


                // TODO revise below as we've already performed unaliasing so that the isAlias check may be removed
                List<Object> modelList = (List<Object>) models.get("models");
                if (modelList != null && !modelList.isEmpty()) {
                    Map<String, Object> modelTemplate = (Map<String, Object>) modelList.get(0);
                    if (modelTemplate != null && modelTemplate.containsKey("model")) {
                        CodegenModel m = (CodegenModel) modelTemplate.get("model");
                        if (m.isAlias && !(config instanceof PythonClientCodegen))  {
                            // alias to number, string, enum, etc, which should not be generated as model
                            // for PythonClientCodegen, all aliases are generated as models
                            continue;  // Don't create user-defined classes for aliases
                        }
                    }
                    allModels.add(modelTemplate);
                }


                // to generate model files
                generateModel(files, models, modelName);


                // to generate model test files
                generateModelTests(files, models, modelName);


                // to generate model documentation files
                generateModelDocumentation(files, models, modelName);
```

Reply:
maybe it is an issue with the mustache file accessing the value 🤔 ? where that line is located is wrapped by `{{#vars}}{{/vars}}` instead of `{{#allVars}}{{/allVars}}`... would that have anything to do with it?

Reply:
In this area I can't help much because because I never touch that area before :/

Reply:
@4brunu do you know anyone who might be able to point us in the right direction?

Reply:
Maybe @wing328 can help with this?

Reply:
@wing328 TL;DR

In `postProcessModels` we inject a custom property vendorExtension called `x-null-encodable-default-value`. However, that value doesn't appear to be present/usable in the `model_doc.mustache` template.

Reply:
@4brunu bumping this incase you have any different thoughts. hope you are having a Merry Christmas season 🎄

Reply:
Hi @jarrodparkes, sorry but I don't have any ideas.
But since this only affects documentation, maybe it's okay to leave it like that

Reply:
that works for me